setup-database.sh

```diff
@@ -8,22 +8,41 @@ SETUP_LOCKFILE="${DATADIR}/.postgresql.init.lock"
 # This script will setup the necessary folder for database
 chown -R postgres /var/lib/postgresql
 # test if DATADIR has content
-if [[ -z "${EXISTING_DATA_DIR}" ]]; then \
-    if [[ ! -f "${SETUP_LOCKFILE}" ]]; then
+if [[ "${EXISTING_DATA_DIR}" == 'FALSE' ]]; then
+    # Only attempt reinitializations if ${DATADIR} doesn't exists
```

Comment:
When PostgreSQL is installed it initializes the data directory at /var/lib/postgresql/12/main. Previously we used to check if the data directory exists and hence the https://github.com/kartoza/docker-postgis/issues/201 which I attempted to solve. 

I think we can either check to see what files exists when a DB is initialized vs an empty DB.
That would be a simpler solution to handle the upgrade situation but still also let first-time users spin up the image without having to check the logs

Reply:
Just CMIIW, in earlier solution new COLLATION is applied by reinitializing a new DB?
Is there no way to change COLLATION without destroying existing one?
If we can find those, then we don't need init lock file because as you said, the data dir was initialized by PostgreSQL from inside the image, even before entrypoint script is running. We can't intercept those.

Reply:
During initialization PostgreSQL creates a directory ie
![du-sh-default-12](https://user-images.githubusercontent.com/2510900/76541163-d2515a00-648b-11ea-9531-0f823cccb092.png)

I think we should be able to figure out how to compare this data directory to the one where a running instance is so that we can determine if it's the first time running the image or it's an existing image

Reply:
@lucernae When Postgres is installed it initializes the data directory and creates the postgresql.conf. When our scripts run it populates postgresql.conf. We could run the following
```

   # Search for custom settings defined in /setup-conf.sh
    if [[ grep -Fxq "xmloption = 'document'" ${CONF} ]]; then
       echo " Postgresql has been run before"
    else
        # No content yet - first time pg is being run!
        # No Replicate From settings. Assume that this is a master database.
        # Initialise db
        echo "Initializing Postgres Database at ${DATADIR}"
        rm -rf ${DATADIR}/*
        chown -R postgres /var/lib/postgresql
        su - postgres -c "$INITDB -U postgres -E ${DEFAULT_ENCODING} --lc-collate=${DEFAULT_COLLATION} --lc-ctype=${DEFAULT_CTYPE} --wal-segsize=${WAL_SEGSIZE} -D ${DATADIR}"
    fi

```
We can even then drop the $SETUP_LOCKFILE and ${EXISTING_DATA_DIR}